### PR TITLE
Add GitHub action go E2E test v0

### PR DESCRIPTION
**Pull request by adshmh**
This PR adds the GH workflow file plus other pieces necessary to run the E2E tests:
- A script to update the E2E config file from Repo's secret(s).
- A new `yml` file under GitHub workflows directory.

### CODE REVIEW

File: e2e/.example.test.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ shannon_config:
     rpc_url: https://shannon-testnet-grove-seed-rpc.poktroll.com # replace with your full node rpc url
     grpc_config:
       host_port: shannon-testnet-grove-seed-grpc.poktroll.com:443 # replace with your full node grpc host/port
-    gateway_address: cf09805c952fa999e9a63a9f434147b0a5abfd10f268879694c6b5a70e1ae177 # replace with your gateway private key
-    gateway_private_key: pokt15vzxjqklzjtlz7lahe8z2dfe9nm5vxwwmscne4 # replace with your gateway address
+    gateway_private_key: cf09805c952fa999e9a63a9f434147b0a5abfd10f268879694c6b5a70e1ae177 # replace with your gateway private key
+    gateway_address: pokt1up7zlytnmvlsuxzpzvlrta95347w322adsxslw # replace with your gateway address
     delegated_app_addresses: # replace with the list of apps that are delegated to the gateway
-      - pokt1d8fae8f295a303d686abd7f17c1584a67a3a4a
+      - pokt1gsufws4c5dhemd0z3njfh859xl0624xyyw5m87
```
Comment (commoddity) [inline e2e/.example.test.yaml:18]: Why the change here?

**Review by commoddity — APPROVED** [review]
LGTM

File: .github/workflows/run-e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      - name: copy E2E config
+        run: make copy_test_config
+
+      - name: update E2E config from secrets
+        env:
+          GATEWAY_PRIVATE_KEY: ${{ secrets.GatewayPrivateKey }}
```
Comment (commoddity) [inline .github/workflows/run-e2e-tests.yml:24]: The secret name in the repo is all caps eg `GATEWAYPRIVATEKEY` so we either need to change it here or change the secret name to PascalCase.

**Review by commoddity — CHANGES_REQUESTED** [review]
See comment above casing of secret name.

File: .github/workflows/run-e2e-tests.yml (hunk 2 above)
Comment (adshmh) [reply .github/workflows/run-e2e-tests.yml]: GH secrets are case-insensitive, but this is a good catch: updated to be consistent with `poktroll`.

**Review by commoddity — APPROVED** [review]
